lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,534 @@
+      pdr_cumulus_id: null,
+      provider_cumulus_id: null,
+      query_fields: null,
+      beginning_date_time: null,
+      ending_date_time: null,
+      last_update_date_time: null,
+      processing_end_date_time: null,
+      processing_start_date_time: null,
+      production_date_time: null,
+      timestamp: null,
+      created_at: new Date(testGranule.createdAt),
+      updated_at: new Date(testGranule.updatedAt),
+    }
+  );
+});
+
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  await t.throwsAsync(
+    migrateGranuleRecord(testGranule, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migrateFileRecord handles nullable fields on source file data', async (t) => {
+  const {
+    filePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testFile = testGranule.files[0];
+
+  delete testFile.bucket;
+  delete testFile.checksum;
+  delete testFile.checksumType;
+  delete testFile.fileName;
+  delete testFile.key;
+  delete testFile.path;
+  delete testFile.size;
+  delete testFile.source;
+
+  const granuleCumulusId = await migrateGranuleRecord(testGranule, knex);
+  await migrateFileRecord(testFile, granuleCumulusId, knex);
+
+  // Also unsure of condition for null bucket and key
+  const record = await filePgModel.get(knex, {});
+
+  t.deepEqual(
+    omit(record, fileOmitList),
+    {
+      bucket: null,
+      checksum_value: null,
+      checksum_type: null,
+      file_size: null,
+      file_name: null,
+      key: null,
+      source: null,
+      path: null,
+    }
+  );
+});
+
+test.serial('migrateGranulesAndFiles skips already migrated granule record', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  await Promise.all(testGranule.files.map((file) => s3Utils.s3PutObject({
+    Bucket: file.bucket,
+    Key: file.key,
+    Body: 'some-body',
+  })));
+  await migrateGranuleRecord(testGranule, knex);
+  await granulesModel.create(testGranule);
+
+  t.teardown(() => {
+    granulesModel.delete(testGranule);
+  });
+
+  const migrationSummary = await migrateGranulesAndFiles(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    filesSummary: {
+      dynamoRecords: 1,
+      failed: 0,
+      skipped: 0,
+      success: 0,
+    },
+    granulesSummary: {
+      dynamoRecords: 1,
+      failed: 0,
+      skipped: 1,
+      success: 0,
+    },
+  });
+
+  const records = await knex(tableNames.granules);
+  t.is(records.length, 1);
+});
+
+test.serial('migrateGranulesAndFiles processes multiple granules and files', async (t) => {
+  const {
+    knex,
+    testCollection,
+    testExecution,
+    testGranule,
+  } = t.context;
+
+  const alternateBucket = cryptoRandomString({ length: 10 });
+  await s3Utils.createBucket(alternateBucket);
```

Comment:
Is this test setup necessary?

Reply:
No, but I wanted to test a granule with a file that referenced a different bucket. I suppose that isn't necessary.

Reply:
[`972c441` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/972c441b1b60661185d3edfb5b73855e77d411e4)